dcrwallet.go

```diff
@@ -289,7 +290,22 @@ func rpcClientConnectLoop(legacyRPCServer *legacyrpc.Server, loader *wallet.Load
 				legacyRPCServer.SetChainServer(chainClient)
 			}
 			if cfg.EnableTicketBuyer {
-				startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				pm, err = startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				if err != nil {
+					log.Errorf("Unable to start ticket buyer: %v", err)
+					go func() {
+						loader.SetWaitFn(nil)
```

Comment:
I think it should synchronously (not in a newly spawned goroutine) remove it before the new ticket purchaser is started then